descwl_coadd/coadd.py

```diff
@@ -562,54 +568,82 @@ def get_psf_exp(
-    psf_offset = get_psf_offset(pos)
-    # print('psf offset:', psf_offset)
 
-    cy, cx = (np.array(psf_image.shape)-1)/2
-    cy += psf_offset.y
-    cx += psf_offset.x
-    psf_crpix = geom.Point2D(x=cx, y=cy)
-    # print('psf crpix:', psf_crpix)
+    psf_dim = psf_image.shape[0]
+
+    psf_bbox = get_psf_bbox(pos=pos, dim=psf_dim)
+    print(psf_dim)
+    print(psf_bbox)
+
+    # pmasked_image = afw_image.MaskedImageF(psf_dim, psf_dim)
+    # pmasked_image.image.array[:, :] = psf_image
+    # pmasked_image.variance.array[:, :] = var
+    # pmasked_image.mask.array[:, :] = 0
+    #
+    # psf_exp = afw_image.ExposureF(pmasked_image)
+    # wcs same as SE exposure
+    psf_exp = afw_image.ExposureF(psf_bbox, wcs)
+    psf_exp.image.array[:, :] = psf_image
+    psf_exp.variance.array[:, :] = var
+    psf_exp.mask.array[:, :] = 0
+    print('exp shape:', psf_exp.image.array.shape)
 
-    cd_matrix = wcs.getCdMatrix(pos)
+    psf_exp.setFilterLabel(exp.getFilterLabel())
+    # psf_exp.setWcs(wcs)
+    detector = DetectorWrapper().detector
+    psf_exp.setDetector(detector)
 
-    psf_wcs = makeSkyWcs(
-        crpix=psf_crpix,
-        crval=coadd_cen_skypos,
-        # crval=coadd_sky_orig,
-        cdMatrix=cd_matrix,
-    )
-    # print('psf wcs')
-    # print(psf_wcs)
+    return psf_exp
 
-    # psf_wcs = wcs.copyAtShiftedPixelOrigin(
-    #     lsst.geom.Extent2D(x=psf_offset.x, y=psf_offset.y)
-    # )
-    # TODO: deal with zeros
 
-    pny, pnx = psf_image.shape
-    pmasked_image = afw_image.MaskedImageF(pny, pnx)
-    pmasked_image.image.array[:, :] = psf_image
-    pmasked_image.variance.array[:, :] = var
-    pmasked_image.mask.array[:, :] = 0
+def get_psf_bbox(pos, dim):
+    """
+    copied from https://github.com/beckermr/pizza-cutter/blob/66b9e443f840798996b659a4f6ce59930681c776/pizza_cutter/des_pizza_cutter/_se_image.py#L708
+    """  # noqa
 
-    psf_exp = afw_image.ExposureF(pmasked_image)
+    # compute the lower left corner of the stamp
+    # we find the nearest pixel to the input (x, y)
+    # and offset by half the stamp size in pixels
+    # assumes the stamp size is odd
+    # there is an assert for this below
 
-    psf_exp.setFilterLabel(exp.getFilterLabel())
-    psf_exp.setWcs(psf_wcs)
-    detector = DetectorWrapper().detector
-    psf_exp.setDetector(detector)
+    x = pos.x
+    y = pos.y
 
-    return psf_exp
+    half = (dim - 1) / 2
+    x_cen = np.floor(x+0.5)
+    y_cen = np.floor(y+0.5)
+
+    # make sure this is true so pixel index math is ok
+    assert y_cen - half == int(y_cen - half)
+    assert x_cen - half == int(x_cen - half)
+
+    # compute bounds in Piff wcs coords
+    xmin = int(x_cen - half)
+    ymin = int(y_cen - half)
+
+    return geom.Box2I(
+        geom.Point2I(xmin, ymin),
+        geom.Point2I(xmin + dim-1, ymin + dim-1),
+    )
+
+
+def get_coadd_psf_bbox(x, y, dim):
+    """
+    suggested by Matt Becker
+    """
+    xpix = int(x)
+    ypix = int(y)
```

Comment:
This only works for integer positions.

Reply:
how would you do it?

(note this isn't the cause of the image being all NaN)

Reply:
The coadd PSF position should always be integers, so the code is fine.